src/middlewares.ts

```diff
@@ -15,37 +15,85 @@ import {
 	touchCommit,
 } from './api';
 
-export async function determineCommitHash(req: any, res: any, next: any) {
-	const isHashInSession = !!req.session.commitHash;
-	const isHashSpecified = req.query && (req.query.hash || req.query.branch);
-	let commitHash;
+function assembleSubdomainUrlForHash(req: any, commitHash: string) {
+    const protocol = (req.secure ? 'https' : 'http');
 
-	if (isHashInSession && !isHashSpecified) {
-		next();
-		return;
-	}
+    return protocol + '://hash-' + commitHash + '.' + stripCommitHashSubdomainFromHost(req.headers.host);
```

Comment:
why do we need the `hash-` prefix?

Reply:
I explained the "hash-" prefix on this commit:
https://github.com/Automattic/dserve/pull/62/commits/8c164c42c8f2bd7ebb64edaf2d53b3e6c2062cf5
Basically, it seems like I need either a predictable/consistent format for the hash subdomain or a configuration file that specifies the base URL for the environment.